test/jdk/java/foreign/TestByteBuffer.java

```diff
@@ -25,7 +25,7 @@
  * @test
  * @modules java.base/sun.nio.ch
  *          jdk.incubator.foreign/jdk.internal.foreign
- * @run testng/othervm -Dforeign.restricted=permit TestByteBuffer
+ * @run testng/othervm --illegal-access=permit -Dforeign.restricted=permit TestByteBuffer
```

Comment:
Can you change java.base/sun.nio.ch to java.base/sun.nio.ch:+open instead? That would avoid the --illegal-access=permit.

Reply:
In fact the new test only checks if everything is aligned like we expect, but it does not test that our mapping returns a memory segment with expected contents.

Reply:
So I tend to make a simple test without reflection that writes a defined sequence of bytes to a file and then maps it checking that the first byte in the mapped segment is from that sequence.

Reply:
Good suggestion!